HumanControls/HumanControls.cpp

```diff
@@ -0,0 +1,137 @@
+                             unsigned int pressureIncrement,
+                             unsigned int pressureMin,
+                             unsigned int pressureMax,
+                             unsigned int durationIncrement,
+                             unsigned int durationMin,
+                             unsigned int durationMax,
+                             unsigned int numButtons,
+                             unsigned int encoderPinSW,
+                             unsigned int enablePin,
+                             unsigned int firePin)
+    : m_menuController(encoderPinClk, encoderPinDt, displayAddress, displayLen, displayWidth,
+                       angleIncrement, angleMin, angleMax, pressureIncrement, pressureMin,
+                       pressureMax, durationIncrement, durationMin, durationMax),
+      m_pinDebouncer(numButtons), m_enableController(), m_fireController() //,
+                                                                           /*encoderPinSW(encoderPinSW), enablePin(enablePin), firePin(firePin)*/
+{
+    this->m_encoderPinSW = encoderPinSW;
+    this->m_enablePin = enablePin;
+    this->m_firePin = firePin;
+    this->m_isConnected = false;
+}
+
+void HumanControls::init()
+{
+    connect();
+    m_pinDebouncer.addPin(this->m_encoderPinSW, HIGH, INPUT_PULLUP);
+    m_pinDebouncer.addPin(this->m_enablePin, HIGH, INPUT_PULLUP);
+    m_pinDebouncer.addPin(this->m_firePin, HIGH, INPUT_PULLUP);
+    m_pinDebouncer.begin();
+    m_menuController.init(status);
+}
+
+void HumanControls::update()
+{
+    if (m_isConnected == false)
+    {
+        this->connect();
+    }
+
+    this->setStatus();
+
+    m_menuController.menuUpdate(status);
+    m_pinDebouncer.update();
+}
+
+void HumanControls::setStatus()
+{
+
+    if (this->m_isConnected)
+    {
+        //Check the robot's status and see if it is different then the controllers.
+        //If it is, set the status to the lowest given status. Ex. Robot returns disabled, but controller enabled, so
+        //set both statuses to disabled. Robot status and controller status should always be the same.
+        //Only exception may be due to firing setting status to enabled after firing.
+
+        if (m_enableController.getIsEnabled())
+        {
+            if (m_fireController.getIsFireToggled())
+            {
+                status = HumanControls::primed;
+            }
+            else
+            {
+                status = HumanControls::enabled;
+            }
+        }
+        else
+        {
+            status = HumanControls::disabled;
+        }
+    }
+    else
+    {
+        status = HumanControls::disabled;
+    }
+
+    if (status != lastStatus)
+    {
+        Serial.println(status);
+        lastStatus = status;
+        m_menuController.menuRefresh(status);
+    }
+
+    //Enventually will set the robot's status here
+}
+
+//Methods for debouncer
+void HumanControls::onPinActivated(int pinNr)
+{
+    if (pinNr == m_encoderPinSW)
```

Comment:
I think this could be a switch/case, but this is fine too.

Reply:
I had it as a switch case, but it kept giving something along the lines of " 'this' is not a constant expression"  when I transferred it to the class. From what I found I have to make it static for it to work. I remember using switch/cases in non-static member methods previously, so I was not sure why it was not working, and eventually just switched it to an if.